p2p/chain.py

```diff
@@ -1,11 +1,14 @@
 import asyncio
 import logging
 import operator
+import random
 import time
 from typing import Any, cast, Dict, List, Set, Tuple  # noqa: F401
 
+from evm.constants import EMPTY_UNCLE_HASH
 from evm.db.chain import AsyncChainDB
 from evm.rlp.headers import BlockHeader
+from evm.utils.state import make_trie_root_and_nodes
```

Comment:
I want to avoid cross-library imports from `utils`.  Possible solutions.

- copy this to a local `p2p.utils`
- create an alternate location for utils that are considered public API (this seems like a good idea).

Reply:
Yeah, makes sense. I've moved it to `evm.db.trie`, unless you have another place in mind?